### PR TITLE
Add recipe for bear

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/bear/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:  # [osx]
```
Comment (wolfv) [inline recipes/bear/conda_build_config.yaml:1]: Is this really necessary, or does `CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"` in the build.sh work?

https://github.com/conda-forge/conda-forge.github.io/pull/1547

File: recipes/bear/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -e
+
+mkdir build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (wolfv) [inline recipes/bear/build.sh:6]: Maybe you can use `${CMAKE_ARGS}`. that will ease cross compilation later on.

File: recipes/bear/conda_build_config.yaml (hunk 1 above)
Comment (mwittgen) [reply recipes/bear/conda_build_config.yaml]: This is not sufficient: 
 no member named 'clock_gettime' in the global namespace

Comment (mwittgen) [reply recipes/bear/conda_build_config.yaml:1]: clock_gettime requires SDK 10.12+

**Comment by mwittgen** [comment]
Looks like this package requires the use of SDK 10.12+ as clock_gettime
is used. Is there a better suggested solution to resolve this?
Other suggestions are implemented.

**Comment by mwittgen** [comment]
@conda-forge/staged-recipes
This PR seems to have stalled.
Use of clock_gettime seems to mandate the use of SDK 10.12
https://conda-forge.org/docs/maintainer/knowledge_base.html

**Pull request merged**